File: src/01-easy/ascii-art/solution.cs
using System;
class Solution {
    static void Main(){
        int L=int.Parse(Console.ReadLine()),H=int.Parse(Console.ReadLine());
        string T=Console.ReadLine().ToUpper();
        string[] rows=new string[H];
        for(int i=0;i<H;i++)rows[i]=Console.ReadLine();
        for(int i=0;i<H;i++){
            string line="";
            foreach(char c in T){
                int idx=c-'A';if(idx<0||idx>25)idx=26;
                line+=rows[i].Substring(idx*L,L);
            }
            Console.WriteLine(line);
        }
    }
}

File: src/01-easy/ascii-art/solution.vb
Imports System
Module Solution
    Sub Main()
        Dim L As Integer=Integer.Parse(Console.ReadLine()),H As Integer=Integer.Parse(Console.ReadLine())
        Dim T As String=Console.ReadLine().ToUpper()
        Dim rows(H-1) As String
        For i As Integer=0 To H-1:rows(i)=Console.ReadLine():Next
        For i As Integer=0 To H-1
            Dim line As String=""
            For Each c As Char In T
                Dim idx As Integer=Asc(c)-65:If idx<0 Or idx>25 Then idx=26
                line &= rows(i).Substring(idx*L,L)
            Next
            Console.WriteLine(line)
        Next
    End Sub
End Module
